guide/src/migration.md

```diff
@@ -3,6 +3,42 @@
 This guide can help you upgrade code through breaking changes from one PyO3 version to the next.
 For a detailed list of all changes, see the [CHANGELOG](changelog.md).
 
+## from 0.18.* to 0.19
+
+### Smarter `anyhow::Error` / `eyre::Report` conversion when inner error is "simple" `PyErr`
+
+When converting from `anyhow::Error` or `eyre::Report` to `PyErr`, if the inner error is a "simple" `PyErr` (with no source error), then the inner error will be used directly as the `PyErr` instead of wrapping it in a new `PyRuntimeError` with the original information converted into a string.
+
+```rust
+# #[cfg(feature = "anyhow")]
+# #[allow(dead_code)]
+# mod anyhow_only {
+# use pyo3::prelude::*;
+use anyhow::Result;
+fn raise_err() -> anyhow::Result<_> {
+    // Here we simulate a simple `PyErr` is raised
+    Err(PyValueError::new_err("Rust Value Error"))?;
+}
+
+fn main() {
+    Python::with_gil(|py| {
+        let res = raise_err();
+
+        // Similar conversion is performed when the function is wrapped with #[pyfunction]
```

Comment:
I think it would be nice to just define a `#[pyfunction]` in the example and call that using `wrap_pyfunction!` and `py_run!` instead of hinting at this in a comment.